### ISSUE TITLE
about ViT architecture

### ISSUE THREAD

**Issue by BeCarefulOfYournaoke** [winci-ai/resa#1]
Congratulations on your work being accepted by ICML. There are ViT results in Table 4 of the paper, and only the architecture of ResNet in this code. Can you provide the code about the ViT architecture? Thanks for your official code.

**Comment by winci-ai** [comment]
Hi, thank you very much for raising the issues.

I've added the training code for the ViT model we used in our paper under this thread. You can also download the pretrained ViT model we provided at: `https://github.com/winci-ai/resa/releases/tag/vit`.

We didn't include the ViT training code in the main repository initially because we’ve been actively exploring better strategies and methods to train ReSA on ViT. We will be updating the repository with these improvements soon.

[code.zip](https://github.com/user-attachments/files/20513941/code.zip)

**Comment by dongyaoli10x** [comment]
Thank you for this very interesting work.

Follow up on this thread - based on the comment above, have you experienced challenges in training ReSA with ViT as backbone encoder? If so, could you elaborate on those challenges and the suspected root causes? Appreciate it!

**Comment by winci-ai** [comment]
Thanks for the questions. 

Yes, training SSL methods with ViTs requires extensive hyperparameter tuning. I previously experimented with the DINO framework and found that ReSA did not converge well in that setting. As a result, I later adopted the training strategy of MoCo v3.

Recently, I integrated ReSA into the DINOv2 framework and observed that, under this setup, ReSA can achieve strong and stable performance when training ViTs. I believe the key reason is that DINOv2 incorporates several techniques that effectively stabilize training, such as LayerScale, layer-wise learning rate decay, and the fused SwiGLU FFN.
